doc/Custom-layouts.md

```diff
@@ -1,112 +1,114 @@
-
-### XML declaration
-
-Every keyboard XML starts with `<?xml version="1.0" encoding="utf-8"?>`.
-
-### Keyboard metadata
-
-The `<keyboard>` tag encloses the whole keyboard. The following properties may be used:
-
-* `name`: The name of the keyboard as it appears in the settings menu. If not present, it will just be called “Custom layout”.
-* `bottom_row`: Whether or not to show the common bottom row. Accepts `true` or `false`, and defaults to `true`.
-* `script`: The (main) writing system that it supports. Possible values are `arabic`, `armenian`, `bengali`, `cyrillic`, `devanagari`, `gujarati`, `hangul`, `hebrew`, `latin`, `persian`, `shavian`, and `urdu`. Defaults to `latin`.
+## Keyboard metadata
+The `<keyboard>`...`</keyboard>` pair follows the declaration tag and encloses the whole keyboard. The following properties may be used (The first two appear in the example above):
+* `name`: The name of the keyboard. The name you specify will appear in the Settings menu. If not present, the layout will just appear as “Custom layout”.
+* `script`: The (main) writing system that the keyboard supports. The possible values are `arabic`, `armenian`, `bengali`, `cyrillic`, `devanagari`, `gujarati`, `hangul`, `hebrew`, `latin`, `persian`, `shavian`, and `urdu`. Defaults to `latin`.
 * `numpad_script`: The script to use for the numpad. This is useful for scripts where a different, non-ASCII set of numerals is used, like Devanagari and Arabic. Defaults to the same as `script`.
+* `bottom_row`: Whether or not to show the common bottom row. It accepts `true` or `false`, and defaults to `true`. If your custom layout defines a bottom row, you should specify `bottom_row=false` to disable the built-in bottom row.
 
-### Modmap
-
-The `<modmap>` tag encloses custom mappings for the Shift and Fn key’s behavior.
+## Row
+The `<row>`...`</row>` pair encloses one row on the keyboard. It has only one optional property:
+* `height`: The height of the row: a positive floating-point value.
 
-Each entry contains two mandatory properties: `a` for the base character before the modifier is applied, and `b` for the modified character.
+A row's standard height is 1.0. The `height` property makes the row taller or shorter than the others. For example, if you define a 5-row keyboard but the top row has `height="0.7"`, then the keyboard's total height is 4.7 units.
 
-For example, to make the `I` key behave as in Turkish:
-
-    <modmap>
-        <shift a="i" b="İ" />
-    </modmap>
-
-There can be as many of these tags inside `<modmap>` as needed.
+You set the physical height of the complete keyboard in Settings as a percentage of the total height of the screen, which can be different between portrait and landscape. Unexpected Keyboard scales the "units" so all the rows fit in that alloted space. Thus, defining a shorter row gives space to the other rows.
```

Comment:
That's not true. The percentage set in the setting is only true for a 4 row layout where every rows have a height of `1.0`. Adding or removing a row or changing a row's height will change the keyboard apparent size and will not scale the keys.

The interaction between unusual layouts and this option has never be thought of so I suggest dropping this paragraph.